packages/common/driver-definitions/src/storage.ts

```diff
@@ -450,3 +453,42 @@ export enum FetchSource {
 	default = "default",
 	noCache = "noCache",
 }
+
+/**
+ * This represents the snapshot tree structure with group IDs. It could be a partial snapshot tree
+ * where the missing tree structure inside another tree could be represented with a groupId and the
+ * actual tree structure could be missing. This group Ids could be then be used to fetch the actual
+ * tree structure from the storage service.
+ * @alpha
+ */
+export interface ISnapshotTree2 extends ISnapshotTree {
+	/**
+	 * Represents the group to which the children belongs to.
+	 */
+	groupID?: string;
```

Comment:
Given that groupId is optional, should this be a chance of ISnapshotTree directly?
This direction would not be ideal / right if client could write that value to service, but I think it's only used service -> client communication, so we are not breaking any protocol contracts with such change. We will break runtime contracts, but I think we are breaking them anyway with this PR

The secondary question - do we have such concept as partial snapshot without content? We probably do (for staging purposes), but I think we should consider option of not having ISnapshotTree2 / changing ISnapshotTree, but rather make all the changes (including introduction of groupID) directly as part of IPartialSnapshotWithContents.

I might be wrong though.  I'm simply trying to understand if we could use less types / concepts.

Reply:
Added groupId directly to ISnapshotTree in another PR and made IPartialSnapshotWithContents extends ISnapshotTree. No there is no concept for IPartialSnapshotTree without contents.

Reply:
@vladsud Why it would be wrong if we are adding when client needs to supply that to service. We would add it to ISummaryTree too, right?